pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -495,6 +503,18 @@ func (r *ReconcileLogStorage) elasticsearchSecrets(ctx context.Context) ([]*core
 	return secrets, nil
 }
 
+// Returns true if we want to apply a new trial license.
```

Comment:
This function comment isn't super useful, since this information is essentially contained in the function name :) 

I'd prefer if there was some more explanation here - i.e., in what circumstances do we expect to need a trial license? What happens if a user wants to switch from a trial license to a non-trial license? etc.